Memoria.Patcher/Program.cs

```diff
@@ -89,27 +90,46 @@ private static void Run(String[] args)
             {
                 Int64 magicNumber;
                 BinaryReader br = new BinaryReader(inputFile);
-                try
-                {
-                    // if the file is not signed
-                    inputFile.Seek(-3 * 8, SeekOrigin.End);
 
-                    magicNumber = br.ReadInt64();
-                    if (magicNumber != 0x004149524F4D454D)// MEMORIA\0 
-                        throw new InvalidDataException("Invalid magic number: " + magicNumber);
+                Assembly assembly = Assembly.GetExecutingAssembly();
+                Module module = assembly.GetModules().First();
+                X509Certificate certificate = module.GetSignerCertificate();
+                // certificates vary slightly in size but are always within a few bys this range is small enough that it's quick but will find the magic number
+                // how to sign this patcher once it's completly built run from Output directory
+                // signtool sign /d "Memoria Patcher for Modding FF9" /td SHA256 /fd SHA256 /sha1 D0E766E285EBE4E93AC723FAE9369495440A67F9 /tr http://timestamp.digicert.com .\Memoria.Patcher.exe 
+                if (certificate != null)
+                {
+                    Console.WriteLine("Memoria.Patcher is Digitally signed, please wait while we validate");
+                    Int64 possition = -0x2920;
+                    bool Found = false;
+                    while (!Found && possition < -0x2700)
+                    {
+                        inputFile.Seek(possition, SeekOrigin.End);
+                        magicNumber = br.ReadInt64();
+                        if (magicNumber == 0x004149524F4D454D)
+                        {
+                            Found = true;
+                            break;
+                        }
+                        possition += 1;
+                    }
+                    if (!Found)
+                        throw new InvalidDataException("File is Signed but could not find magic number, file is corrupt please notify a Memoria Team Member.");
                 }
-                catch (Exception ex)
+                else
                 {
-                    // if the file is signed this offset might change slightly due to the way signing works.
-                    // you will need to use a Hex editor to make sure that from the M of MEMORIA to the end of the file
-                    // is length you offset this by
-                    inputFile.Seek(-0x28F5, SeekOrigin.End);
+                    // Confirm! do we want this?
```

Comment:
Please review should we enable these console logs it means if we do releases without signing it will warn users, me and snouz are looking for other developer feedback on this matter?

Reply:
I agree with Snouz that a message like that is not a good thing.
We have done many releases that were not signed and were still official versions; it may still happen in the future and then that message would be false.
Also, the point of signing patcher is to make Windows trust the program, not the users. A non-trustworthy version of the patcher could easily be compiled without that message anyways, so I don't think that adds any more security from the user's end.

Reply:
okays, i will remove it then